src/js/fs.js

```diff
@@ -20,6 +20,11 @@ var util = require('util');
 var fsBuiltin = native;
 
 fs.exists = function(path, callback) {
+  if ((!(util.isString(path)) && !(util.isBuffer(path)))
```

Comment:
I'm not sure this is a good solution: shouldn't the error be reported via the callback?

Reply:
@galpeter I think we must consider the case when user does not pass callback ( It is strange, but it is possible). In this case, we have no way to inform the user there is an error since there is no callback. Node.js seems throwing TypeError for this case. Here is my test.

```sh
$ node -e 'fs = require('fs');fs.exists(3.14159);'

assert.js:81
  throw new assert.AssertionError({
  ^
AssertionError: path must be a string
    at Module.require (module.js:496:3)
    at require (internal/module.js:20:19)
    at [eval]:1:6
    at ContextifyScript.Script.runInThisContext (vm.js:25:33)
    at Object.runInThisContext (vm.js:97:38)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:570:32)
    at evalScript (bootstrap_node.js:354:27)
    at run (bootstrap_node.js:123:11)
    at run (bootstrap_node.js:390:7)
```
If I misunderstand what you mean, could you explain in code what do you mean by the error reported via the callback?

Reply:
@glistening I was unsure how the node works. That is for example: ` fs.exists(3, function(er) { .. });` will the error be reported via callback or not. But your example clarifies that such error is always reported in a sync fashion.

So this solution looks good.